internalshared/configutil/listener.go

```diff
@@ -410,7 +411,16 @@ func ParseListeners(result *SharedConfig, list *ast.ObjectList) error {
 
 // ParseSingleIPTemplate is used as a helper function to parse out a single IP
 // address from a config parameter.
+// If the input doesn't appear to contain the 'template' format,
+// it will return the specified input unchanged.
 func ParseSingleIPTemplate(ipTmpl string) (string, error) {
+	m, err := regexp.MatchString("(.*?)({{.*?}})(.*)", ipTmpl)
```

Comment:
```suggestion
	m, err := regexp.MatchString("{{.*?}}", ipTmpl)
```

We just need to know that somewhere in the input there's a `{{` followed eventually by a `}}`, right?

Reply:
Also check out regexp.MustCompile.  Any chance you get to turn a run-time error into a compile-time error is usually worth taking.